### PR TITLE
task 6 on branch6 complted with tests

### PR DESCRIPTION
**Pull request by Szpytma**

### CODE REVIEW

**Review by alexdevdavis — CHANGES_REQUESTED** [review]
Here are some extra tests to add before this can be considered done:
- when review_id is invalid (status 400)
- when review_id valid but not found (status 404)
- when the review exists, but has no comments (status 200)


Perhaps you've written some of these in the past; check your commit history if you feel you have, as you may easily be able to add them to this branch and save a bunch of time

**Comment by Szpytma** [comment]
@alexdevdavis 
when review_id is invalid (status 400) - completed
when review_id valid but not found (status 404) - completed
when the review exists, but has no comments (status 200) - I guess I need to create a func checking if the review exists and then grab all promises - something as we did on today lecture, I was trying to complete it today but I was struggling

File: __tests__/app.test.js [hunk 1]
```diff
@@ -135,3 +135,54 @@ describe("GET /api/reviews/:review_id", () => {
...
+        expect(body.message).toBe("Please provide an valid ID");
+      });
+  });
+
+  it("404: responds with an error when review_id valid but not found", () => {
+    return request(app)
+      .get("/api/reviews/9999/comments")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.message).toBe("Index outOfBound");
```
Comment (alexdevdavis) [inline __tests__/app.test.js]: I wonder if `review not found` might be a more useful message

File: models/reviews.model.js [hunk 2]
```diff
@@ -16,3 +16,27 @@ exports.fetchAllReviews = () => {
...
+    }
+    return comments.rows;
+  });
+};
+
+const checkReviewExist = (id) => {
+  let queryStr = `SELECT * FROM reviews WHERE review_id = $1;`;
+  return db.query(queryStr, [id]).then((result) => {
+    if (result.rowCount === 0) {
+      return Promise.reject({ status: 404, message: "Index outOfBound" });
```
Comment (alexdevdavis) [inline models/reviews.model.js]: No sure where Index outOfBound has come from, but it doesn't reflect what the user is trying to do, so I'd suggest changing it

**Review by alexdevdavis — APPROVED** [review]
Happy to approve this PR. Well done 😎 
- just the one error message to edit, and it's 💯 . No need to submit another PR for this ticket, though; I'm very happy with the work you've done here

**Pull request merged**